ESP32-C6 wifi build failure if CONFIG_ESP32_WIFI_MBEDTLS_CRYPTO=y set

### Describe the bug

Setting CONFIG_ESP32_WIFI_MBEDTLS_CRYPTO=y causes a build failure due to a missing file ${MBEDTLS_REMOVED_MODULES_PATH}/ecdh.c

I don't seem to find a ecdh.c in the mbedtls module that ships with Zephyr 4.4. Just removing the target_sources file seems to build and work, maybe the file is no longer needed?

### Regression

- [ ] This is a regression.

### Steps to reproduce

1. Enable WIFI and CONFIG_ESP32_WIFI_MBEDTLS_CRYPTO=y on a ESP32-C6
2. Build
3. Build fails

### Relevant log output

```shell
CMake Error at <project-ws>/zephyr/modules/mbedtls/legacy_support.cmake:116 (target_sources):
  Cannot find source file:

    /port/mbedtls/removed/ecdh.c
Call Stack (most recent call first):
<project-ws>/zephyr/modules/mbedtls/CMakeLists.txt:75 (include)
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- Linux
- zephyr-sdk-1.0.1
- Zephyr 4.4.0

### Additional Context

Keeping CONFIG_ESP32_WIFI_MBEDTLS_CRYPTO=y set but removing the line "target_sources(builtin PRIVATE ${MBEDTLS_REMOVED_MODULES_PATH}/ecdh.c)" does seem to build and wifi AP works with WPA2, maybe the needed private got moved into some other file that is still being built, or are no longer needed? Also not sure if WPA2 uses ecdh, so maybe I'm not using the missing symbols.

--- a/drivers/wifi/esp32/Kconfig.esp32
+++ b/drivers/wifi/esp32/Kconfig.esp32
@@ -380,24 +380,24 @@ config ESP32_WIFI_MBEDTLS_CRYPTO
 	select PSA_WANT_ALG_CMAC
 	select PSA_WANT_ALG_CTR
 	select PSA_WANT_ALG_ECB_NO_PADDING
+	select PSA_WANT_ALG_ECDH
+	select PSA_WANT_ALG_ECDSA
 	select PSA_WANT_ALG_HMAC
 	select PSA_WANT_ALG_SHA_1
 	select PSA_WANT_ALG_SHA_256
+	select PSA_WANT_ECC_SECP_R1_256
 	select PSA_WANT_KEY_TYPE_AES
+	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_BASIC
+	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_IMPORT
+	select PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY
 	help
 	  Select this option to use MbedTLS crypto APIs which utilize hardware acceleration.
 
 config ESP32_WIFI_ENABLE_WPA3_SAE
 	bool "WPA3-Personal"
 	select ESP32_WIFI_MBEDTLS_CRYPTO
-	select PSA_WANT_ALG_ECDH
-	select PSA_WANT_ALG_ECDSA
-	select PSA_WANT_ECC_SECP_R1_256
-	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_BASIC
 	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_EXPORT
 	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_GENERATE
-	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_IMPORT
-	select PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY
 	help
 	  Select this option to allow the device to establish a WPA3-Personal connection.
 

